mycroft/gui/extensions.py

```diff
@@ -0,0 +1,243 @@
+import threading
+from mycroft.configuration import Configuration
+from mycroft.messagebus import Message
+from mycroft.util.log import LOG
+from mycroft.api import is_paired
+from ovos_utils.system import system_reboot, system_shutdown, ssh_enable, ssh_disable
+
+from json_database import JsonStorageXDG
+from mycroft.gui.homescreen import HomescreenManager
+
+
+class ExtensionsManager():
+    def __init__(self, name, bus, gui):
+        """ Constructor for the Extension Manager. The Extension Manager is responsible for
+        managing the extensions that define additional GUI behaviours for specific platforms.
+
+        Args:
+            name: Name of the extension manager
+            bus: MessageBus instance
+            gui: GUI instance
+        """
+
+        self.name = name
+        self.bus = bus
+        self.gui = gui
+        config = Configuration.get()
+        enclosure_config = config.get("enclosure")
+        self.active_extension = enclosure_config.get("extension", "Generic")
+
+        # ToDo: Add Exclusive Support For "Desktop", "Mobile" Extensions
+        self.supported_extensions = ["SmartSpeaker", "Bigscreen", "Generic"]
+
+        if self.active_extension not in self.supported_extensions:
+            self.active_extension = "Generic"
+
+        LOG.info(
+            f"Extensions Manager: Initializing {self.name} with active extension {self.active_extension}")
+        self.activate_extension(self.active_extension)
+
+    def activate_extension(self, extension_id):
+        LOG.info(f"Extensions Manager: Activating Extension {extension_id}")
+
+        # map extension_id to class
+        if extension_id == "SmartSpeaker":
+            self.extension = SmartSpeakerExtension(self.bus, self.gui)
+        elif extension_id == "Bigscreen":
+            self.extension = BigscreenExtension(self.bus, self.gui)
+        else:
+            self.extension = GenericExtension(self.bus, self.gui)
+
+        LOG.info(f"Extensions Manager: Activated Extension {extension_id}")
+        self.bus.emit(
+            Message("extension.manager.activated", {"id": extension_id}))
+
+
+class SmartSpeakerExtension():
+    """ Smart Speaker Extension: This extension is responsible for managing the Smart Speaker
+    specific GUI behaviours. This extension adds support for Homescreens and Homescreen Mangement.
+
+    Args:
+        name: Name of the extension manager
+        bus: MessageBus instance
+        gui: GUI instance
+    """
+
+    def __init__(self, bus, gui):
+        LOG.info("SmartSpeaker: Initializing")
+
+        self.bus = bus
+        self.gui = gui
+        self.homescreen_manager = HomescreenManager(self.bus, self.gui)
+
+        self.homescreen_thread = threading.Thread(
+            target=self.homescreen_manager.run)
+        self.homescreen_thread.start()
+
+        self.device_paired = is_paired()
+        store_conf = ('skill_conf.json')
+
+        self.skill_conf = JsonStorageXDG(store_conf)
```

Comment:
can you explain this please? I'm not sure i understand the point of a new config file here

Reply:
The config stores if the backend selected is local / remote (selene) this is due to the fact that we want to not show the homescreen at all when local backend is selected unless the device reboots on completion of the paring process. Maybe the "skill_conf" sounds like a wrong file name, but its required for knowing what backend has been selected and running through out, unless the setup skill is invoked again manually by the user and he changes the backend.

Reply:
i dont like this, it feels like a bug in the skill that should be solved there and not here.... its a skill specific bug not core specific

usage of override_idle should ensure homescreen doesnt show up when it should not

apart from override_idle, is the issue here the pairing check? should not report paired until the reboot, meaning the homescreen doesnt show and mycroft.ready is not sent (assuming config requires pairing for ready signal)

Reply:
Fixed and Removed